sematic/api/endpoints/runs.py

```diff
@@ -431,17 +431,24 @@ def save_graph_endpoint(user: Optional[User]):
     graph = flask.request.json["graph"]
 
     runs = [Run.from_json_encodable(run) for run in graph["runs"]]
+
     for run in runs:
         logger.info("Graph update, run %s is in state %s", run.id, run.future_state)
+
+        if user is not None:
+            run.user_id = user.id
```

Comment:
I think ownership belongs with the resolution rather than the run. All runs in a given resolution will be owned by the same user.

Reply:
I added a `user_id` column to both `runs` and `resolutions`. This will make it much easier to query. No need to always join to resolutions.